src/browser/modules/Segment/index.tsx

```diff
@@ -0,0 +1,164 @@
+import { updateData } from 'shared/modules/udc/udcDuck'
+
+export interface MetricsProperties {
+  [key: string]: string | number | Date | boolean
+}
+
+export interface MetricsData {
+  category: string
+  label: string
+  data: MetricsProperties
+}
+
+export class Segment extends Component<any> {
+  componentDidMount() {
+    const {
+      segmentKey,
+      setTrackCallback,
+      inDesktop,
+      updateData,
+      children, // eslint-disable-line
+      ...otherProps
+    } = this.props
+    if (!segmentKey || !canUseDOM()) {
+      return
+    }
+    if (!(window as any).analytics) {
+      ;(function(window: any, document: Document, segmentKey: string, a?: any) {
+        const analytics = (window.analytics = window.analytics || [])
+        if (!analytics.initialize) {
+          if (analytics.invoked) {
+            window.console &&
+              console.error &&
+              console.error('Segment snippet included twice.')
+          } else {
+            analytics.invoked = !0
+            analytics.methods = [
+              'trackSubmit',
+              'trackClick',
+              'trackLink',
+              'trackForm',
+              'pageview',
+              'identify',
+              'reset',
+              'group',
+              'track',
+              'ready',
+              'alias',
+              'debug',
+              'page',
+              'once',
+              'off',
+              'on',
+              'addSourceMiddleware',
+              'addIntegrationMiddleware',
+              'setAnonymousId',
+              'addDestinationMiddleware'
+            ]
+            analytics.factory = function(t: any) {
+              return function() {
+                const e = Array.prototype.slice.call(arguments)
+                e.unshift(t)
+                analytics.push(e)
+                return analytics
+              }
+            }
+            for (let t = 0; t < analytics.methods.length; t++) {
+              const e = analytics.methods[t]
+              analytics[e] = analytics.factory(e)
+            }
+            analytics.load = function(t: any, e: any) {
+              const n = document.createElement('script')
+              n.type = 'text/javascript'
+              n.async = !0
+              n.src =
+                'https://cdn.segment.com/analytics.js/v1/' +
+                t +
+                '/analytics.min.js'
+              a = document.getElementsByTagName('script')[0]
+              a.parentNode.insertBefore(n, a)
+              analytics._loadOptions = e
+            }
+            analytics._writeKey = segmentKey
+            analytics.SNIPPET_VERSION = '4.13.2'
+            analytics.load(segmentKey)
+            const doTrack = (metricsData: MetricsData) => {
+              const { category, label, data } = metricsData
+              window.analytics.track(category + '-' + label, {
+                ...data,
+                desktop: inDesktop
```

Comment:
@oskarhane I was thinking we would set the environment to WEB, CLOUD, or DESKTOP?

Reply:
yep, but with the new centrally hosted Browser we will never have `CLOUD` as an `env` anymore once/if Aura move over to using the centrally hosted Browser...

Reply:
Hmm, seems env will not tell us if we're on aura anymore, so rather add a connected to aura 🤔

Reply:
{
hostedInDesktop: boolean,
connectedToAuraHost: boolean
}

Reply:
@OskarDamkjaer @jharris4 
Maybe not part of this PR, but it's a good point that it might be interesting to know.
But the envs would be something like STANDALONE, EMBEDDED, DESKTOP where EMBEDDED it when it's served up from Neo4j server.
Could be worth a card to think about if it's useful and how to identify the differents envs.